### PR TITLE
[Improvement] state/rds-postgres - Add support for parameter and option groups

### PR DESCRIPTION
**Pull request by ambsw-technology**
If you want to [force_ssl](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_PostgreSQL.html#PostgreSQL.Concepts.General.SSL.Requiring) on RDS Postgres, you have to do so by providing a `DBParameterGroup`.  This patch add the option to pass such a group value (with the backwards-compatible default) to the `state/rds-postgres` template.

Obviously, this would work with any parameter group (even those unrelated to `force_ssl`), but here's an example with `force_ssl`:

```
  ForceSSLParameterGroup:
    Type: 'AWS::RDS::DBParameterGroup'
    Properties:
      Description: 'Applies rds.force_ssl to pgSQL servers.'
      Family: 'postgres10'
      Parameters:
        rds.force_ssl: '1'
  Database:
    Type: 'AWS::CloudFormation::Stack'
    Properties:
      Parameters:
        ...
        DBParameterGroupName: !Ref ForceSSLParameterGroup
      TemplateURL: './aws-cf-templates/state/rds-postgres.yaml'
```

### CODE REVIEW

File: state/rds-postgres.yaml [hunk 1]
```diff
@@ -103,6 +103,10 @@ Parameters:
     Type: String
     Default: true
     AllowedValues: [true, false]
+  DBParameterGroupName:
```
Comment (michaelwittig) [inline state/rds-postgres.yaml:106]: Please don't forget to add the new parameter to the meat data at the top as well.

File: state/rds-postgres.yaml [hunk 2]
```diff
@@ -191,6 +196,7 @@ Resources:
       MasterUsername: !If [HasDBSnapshotIdentifier, !Ref 'AWS::NoValue', !Ref DBMasterUsername]
       MasterUserPassword: !If [HasDBSnapshotIdentifier, !Ref 'AWS::NoValue', !Ref DBMasterUserPassword]
       MultiAZ: !Ref DBMultiAZ
+      DBParameterGroupName: !If [HasDBParameterGroupName, !Ref DBParameterGroupName, !Ref 'AWS::NoValue']
```
Comment (michaelwittig) [inline state/rds-postgres.yaml:199]: I like the idea. The same approach would work for `OptionGroupName`. Are you interested in adding this as well?

**Comment by ambsw-technology** [comment]
I pushed another commit w/ `OptionGroup` and rebased on master.  Not sure why it's not showing up here.

**Comment by ambsw-technology** [comment]
I rebased back onto 11.10 (which should help with CI test) and it looks like the extra commit is showing up.

**Comment by michaelwittig** [comment]
I ported this enhancement to the `state/rds-mysql` template: 404ee8861cfac478cea92f8d168d1fdbf903b33b

**Pull request merged**